### ISSUE TITLE
Runworker crashes without traceback when processing background tasks.

### ISSUE THREAD

**Issue by ChillarAnand** [django/channels#1067]
OS

```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04 LTS
Release:        18.04
Codename:       bionic
```

Python packages
```
channels==2.1.0
channels-redis==2.1.1
daphne==2.1.0
Django==1.11.5
Twisted==17.9.0 
channels-redis==2.1.1
```

As mentioneed here https://channels.readthedocs.io/en/latest/topics/worker.html, I am testing background tasks.

Sending event like this

```py
    async_to_sync(channel_layer.send)(
        "send-email",
        {
            'type': 'email',
            'id': 12,
            'message': 'message',
        }
    )
```

These are getting passed to redis 

```sh
127.0.0.1:6379> LRANGE asgi:send-email 0 -1
1) "\x83\xa4type\xa5email\xa2id\x0c\xa7message\xa7message"
2) "\x83\xa4type\xa5email\xa2id\x0c\xa7message\xa7message"
```

Consumer
```py
class SendEmail(SyncConsumer):

    def email(self, message):
        print("Test: " + message)
```

I am running `runworker` locally on my system.

```
$ ./manage.py runworker --traceback -v 3 send-email
2018-05-30 15:40:22,637 - INFO - runworker - Running worker for channels ['send-email']
```

This worker crashes silently wihtout any traceback or error message.

**Comment by andrewgodwin** [comment]
I can't fix a bug without steps to reproduce or more information - is there an exit code the crash leaves behind? Can you `strace` it to see what it does in its final moments?

**Comment by ChillarAnand** [comment]
Thanks @andrewgodwin. It is exiting with code 0.

Here is the relevant part of strace. I am not sure what is happening here.

```
23132 recvfrom(11, "*2\r\n$15\r\nasgi:send-email\r\n$32\r\n\203"..., 262144, 0, NULL, NULL) = 65
23132 rt_sigaction(SIGINT, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7fe548cb3890}, {sa_handler=0x444a20, sa_mask=[], sa_fl
ags=SA_RESTORER, sa_restorer=0x7fe548cb3890}, 8) = 0
23132 mmap(NULL, 262144, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe515daf000

# munmap
23132 sigaltstack(NULL, {ss_sp=0xd1a6c0, ss_flags=0, ss_size=8192}) = 0
23132 sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}, NULL) = 0
```

Complete log is [available here](https://paste.ubuntu.com/p/3gQh7CyBBS/).

**Comment by ChillarAnand** [comment]
Runserver is working fine. There are bunch of websockets which are working fine.

I have just created this test channel and it is failing. I will try out other channels.

**Comment by ChillarAnand** [comment]
It is exiting in the same way for other channels also.

**Comment by andrewgodwin** [comment]
What's your `CHANNEL_LAYERS` setting?

**Comment by ChillarAnand** [comment]
```
ASGI_APPLICATION = "avilpage.routing.application"

CHANNEL_LAYERS = {
    'default': {
        'BACKEND': 'channels_redis.core.RedisChannelLayer',
        'CONFIG': {
            "hosts": [('127.0.0.1', 6379)],
        },
    },
}
```

**Comment by andrewgodwin** [comment]
I really don't know what's happening here - I can't make it fail the same way on my system. Have you tried it on other machines too?

**Comment by ChillarAnand** [comment]
I will try on other systems or create a small project to reproduce the issue.

**Comment by ChillarAnand** [comment]
I think there was a typo in ProtocolType router which is causing the issue.

```
application = ProtocolTypeRouter({
    'chanel': ChannelNameRouter({
        'email-book': consumers.EmailBook,
    }),
})
```

When key is something else instead of `channel`, this issue is happening.

**Comment by andrewgodwin** [comment]
Aha! That might do it, let me check.

**Comment by andrewgodwin** [comment]
I can't seem to replicate that - when I do the same locally, I get:
```
 ./manage.py runworker foo
2018-06-08 22:45:48,418 - INFO - runworker - Running worker for channels ['foo']
Task exception was never retrieved
future: <Task finished coro=<Worker.listener() done, defined at /mnt/c/Users/Andrew/Programs/channels/channels/worker.py:32> exception=ValueError("No application configured for scope type 'channel'",)>
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/mnt/c/Users/Andrew/Programs/channels/channels/worker.py", line 42, in listener
    instance_queue = self.get_or_create_application_instance(channel, scope)
  File "/mnt/c/Users/Andrew/Programs/asgiref/asgiref/server.py", line 87, in get_or_create_application_instance
    application_instance = self.application(scope=scope)
  File "/mnt/c/Users/Andrew/Programs/channels/channels/routing.py", line 58, in __call__
    raise ValueError("No application configured for scope type %r" % scope["type"])
ValueError: No application configured for scope type 'channel'```

**Comment by ChillarAnand** [comment]
I have setup a [small project](https://github.com/ChillarAnand/library) which has this issue. Can you try it out?

**Comment by andrewgodwin** [comment]
Yup, I can reproduce the error with that, thanks!

**Comment by andrewgodwin** [comment]
OK, so it turned out in my case that the error was that the channel layer wasn't connecting to the redis server, but the error was being suppressed so it just looked like it was exiting. I've fixed this in ae6025a, and once I did that it works as intended - could you confirm that this was also your problem?

**Comment by ChillarAnand** [comment]
Yes, it was the problem. Thanks for taking time to troubleshoot and fix it. :-)
